libwget/bar.c

```diff
@@ -188,8 +190,22 @@ wget_bar_t *wget_bar_init(wget_bar_t *bar, int nslots, int max_width)
 	return NULL;
 }
 
+void wget_bar_set_slots(wget_bar_t *bar, int nslots)
+{
+	char lf[nslots];
+    memset(lf, '\n', sizeof(lf));
+
+	if (nslots <= bar->nslots)
+		return;
+	/* _bar_print_slot(bar, 0); */
+	fwrite(lf, 1, nslots - bar->nslots, stdout);
+	bar->nslots = nslots;
+	wget_bar_update(bar);
+}
+
 void wget_bar_register(wget_bar_t *bar, wget_bar_ctx *ctx)
 {
+	assert (ctx->slotpos <= bar->nslots);
```

Comment:
I don't like assertions in library code. Please check for an error condition and handle it gracefully (e.g. print out an error and avoid stopping/crashing the program).